floris/simulation/wake_deflection/gauss.py

```diff
@@ -373,145 +432,116 @@ def calculate_transverse_velocity(
+    V2, W2 = _calculate_vortex(
+        z, HH, D, yLocs, eps, Gamma_bottom, decay=decay, which="bottom", with_decay=True
+    )
 
     # wake rotation vortex
-    zC = z - HH + BaseModel.NUM_EPS
-    rC = yLocs ** 2 + zC ** 2
-    core_shape = 1 - np.exp(-rC / (eps ** 2))
-    V5 = (Gamma_wake_rotation * zC) / (2 * np.pi * rC) * core_shape * decay
-    W5 = (-1 * Gamma_wake_rotation * yLocs) / (2 * np.pi * rC) * core_shape * decay
-
+    V5, W5 = _calculate_vortex(
+        z, HH, D, yLocs, eps, Gamma_wake_rotation, decay=decay, which="rotation", with_decay=True
+    )
 
-    ### Boundary condition - ground mirror vortex
+    # Boundary condition - ground mirror vortex
 
     # top vortex - ground
-    zTb = z + (HH + D / 2) + BaseModel.NUM_EPS
-    rTb = yLocs ** 2 + zTb ** 2
-    # This looks like spanwise decay;
-    # it defines the vortex profile in the spanwise directions
-    core_shape = 1 - np.exp(-rTb / (eps ** 2))
-    V3 = (-1 * Gamma_top * zTb) / (2 * np.pi * rTb) * core_shape * decay
-    W3 = (Gamma_top * yLocs) / (2 * np.pi * rTb) * core_shape * decay
+    V3, W3 = _calculate_vortex(
+        z, HH, D, yLocs, eps, Gamma_top, decay=decay, which="top", with_decay=True, ground=True
+    )
 
     # bottom vortex - ground
-    zBb = z + (HH - D / 2) + BaseModel.NUM_EPS
-    rBb = yLocs ** 2 + zBb ** 2
-    core_shape = 1 - np.exp(-rBb / (eps ** 2))
-    V4 = (-1 * Gamma_bottom * zBb) / (2 * np.pi * rBb) * core_shape * decay
-    W4 = (Gamma_bottom * yLocs) / (2 * np.pi * rBb) * core_shape * decay
+    V4, W4 = _calculate_vortex(
+        z,
+        HH,
+        D,
+        yLocs,
+        eps,
+        Gamma_bottom,
+        decay=decay,
+        which="bottom",
+        with_decay=True,
+        ground=True,
+    )
 
     # wake rotation vortex - ground effect
-    zCb = z + HH + BaseModel.NUM_EPS
-    rCb = yLocs ** 2 + zCb ** 2
-    core_shape = 1 - np.exp(-rCb / (eps ** 2))
-    V6 = (-1 * Gamma_wake_rotation * zCb) / (2 * np.pi * rCb) * core_shape * decay
-    W6 = (Gamma_wake_rotation * yLocs) / (2 * np.pi * rCb) * core_shape * decay
-
-    # total spanwise velocity
-    V = V1 + V2 + V3 + V4 + V5 + V6
-    W = W1 + W2 + W3 + W4 + W5 + W6
-
-    # no spanwise and vertical velocity upstream of the turbine
-    # V[delta_x < -1] = 0.0  # Subtract by 1 to avoid numerical issues on rotation
-    # W[delta_x < -1] = 0.0  # Subtract by 1 to avoid numerical issues on rotation
-    # TODO Should this be <= ? Shouldn't be adding V and W on the current turbine?
-    V[delta_x < 0.0] = 0.0  # Subtract by 1 to avoid numerical issues on rotation
-    W[delta_x < 0.0] = 0.0  # Subtract by 1 to avoid numerical issues on rotation
+    V6, W6 = _calculate_vortex(
+        z,
+        HH,
+        D,
+        yLocs,
+        eps,
+        Gamma_wake_rotation,
+        decay=decay,
+        which="rotation",
+        with_decay=True,
+        ground=True,
+    )
+
+    V = np.add.reduce((V1, V2, V3, V4, V5, V6))
```

Comment:
@RHammond2 Can you explain the benefit us using `np.add.reduce` with the array sequence instead of `np.add` or the np.array `+` operator?

Reply:
At this point, I can't remember why that's there. My hunch is that it was faster, or more convenient for use case. However, after testing out the snippet on small and large arrays, I found it's quite a bit slower compared to `V1 + V2 + V3 + V4 + V5 + V6`. The only reason to use `np.add.reduce` instead of `np.add` is to ensure the addition is handled consistently.

My vote is to put it back to original arithmetic, and if whatever I was attempting to do at that point comes back to me, it can be handled then.

Reply:
Ok cool, thanks for following up. I'll revert that now but we can keep it on our radar for the future